pkg/agent/packetcapture/capture/bpf.go

```diff
@@ -95,7 +119,7 @@ func calculateSkipFalse(srcPort, dstPort uint16) uint8 {
 }
 
 // Generates IP address and port matching instructions
-func compileIPPortFilter(srcAddrVal, dstAddrVal uint32, size, curLen uint8, srcPort, dstPort uint16, needsOtherTrafficDirectionCheck bool) []bpf.Instruction {
+func compileIPPortFilter(srcAddrVal, dstAddrVal uint32, size, curLen uint8, srcPort, dstPort uint16, messages []message, needsOtherTrafficDirectionCheck bool) []bpf.Instruction {
```

Comment:
The name of the function doesn't match the implementation any more. I think we need to switch to a more generic way to provide filters. But maybe we should first wait for the TCP flags PR to be merged so we can rebase. Otherwise we are likely to end up with many conflicts.